### ISSUE TITLE
IoU

### ISSUE THREAD

**Issue by ebgoldstein** [Doodleverse/doodleverse_utils#6]
**Problem:**
IoU metrics reported when training a gym model come from a formulation in doodleverse_utils - `mean_iou()`, which can be found here:

https://github.com/Doodleverse/doodleverse_utils/blob/691abe7f3c3d354df50c2958cf727c4037950222/doodleverse_utils/model_imports.py#L951-L979 

L974 and 975 are what i want to focus on. The tensor is [Batch x Width x Height x NumClasses]. To me, these lines seem to only use the first class to make the IoU metric. 

The effect is that mean IoU metrics being reported are just IoU of the first class. 

**Solution:**

We replace this homebrewed metric with `tf.keras.metrics.OneHotMeanIoU(num_classes = nclasses)`.
so this would be a deletion in doodleverse_utils, and replacing all appearances of `mean_iou` in segmentation_gym

I have tested this solution and it works... 
Furthermore, i have tested to see if the behavior i mention above (mean_iou is just the first class) by comparing its output to: `tf.keras.metrics.OneHotIoU(num_classes= num_classes, target_class_ids = [0])`.. the numbers are nearly identical. 

See the the last cell of the notebook here
https://github.com/ebgoldstein/FRF_GrainSize/blob/main/src/TestModelonC6.ipynb

@dbuscombe-usgs, let me know...

**Comment by ebgoldstein** [comment]
to preserve a homebrewed metric, we could copy from the dice formulation -> https://github.com/Doodleverse/doodleverse_utils/issues/7

so:
```

def new_iou(y_true, y_pred, smooth):
    y_true_f = tf.reshape(tf.dtypes.cast(y_true, tf.float32), [-1])
    y_pred_f = tf.reshape(tf.dtypes.cast(y_pred, tf.float32), [-1])
    intersection = tf.reduce_sum(y_true_f * y_pred_f)
    union =  tf.reduce_sum(y_true_f + y_pred_f) - intersection
    return intersection/union

def new_iou_multi_metric(nclasses, smooth):
    
    def new_iou_coef_multilabel(y_true, y_pred):
        iousum = 0
        y_pred = tf.one_hot(tf.argmax(y_pred, -1), 4)
        for index in range(nclasses):
            iousum += new_iou(y_true[:,:,:,index], y_pred[:,:,:,index], smooth)
        return iousum/nclasses
    
    return new_iou_coef_multilabel
```

**Comment by ebgoldstein** [comment]
i forget that tf.keras.metrics.OneHotMeanIoU does not work for some reason.. so i replaced with the custom metric above.. 
model is running now as a test..

**Comment by ebgoldstein** [comment]
![NOAA_NewLoss_trainhist_8](https://user-images.githubusercontent.com/5330599/193153874-da2b3407-32b9-47fd-b819-be5b6892391a.png)

**Comment by ebgoldstein** [comment]
see more notes in #7 ...
